lib/matplotlib/offsetbox.py

```diff
@@ -1508,7 +1510,33 @@ def __init__(self, ref_artist, use_blit=False):
         disconnect.args[0] for disconnect in self._disconnectors[:2]])
 
     def on_motion(self, evt):
-        if self._check_still_parented() and self.got_artist:
+        # Only check if the widget lock is available, setting it would prevent
+        # picking.
+        if not (
+            self._check_still_parented()
+            and self.canvas.widgetlock.available(self)
+            and self.ref_artist.pickable()
+        ):
+            return
```

Comment:
Is the logic correct? If self.ref_artist.pickable() is False, we return. That means we never reach the original `if self.got_artist:` branch.